### DISCUSSION TITLE
Mixed Integer implementation of OSQP

### DISCUSSION THREAD

**Discussion by Laksh-kit** [osqp/osqp#687] [Matlab/Simulink Interface]
Hello, 

Is it already possible to use the OSQP libraries in Matlab to solve mixed integer problems (MIQCPs)? If not where could the library be extended to make it compatible for MIQCPs. 

Regards
Laksh

**Comment by imciner2** [comment]
Hi @Laksh-kit, we do not support mixed-integer problems in the OSQP library, unfortunately. You can use OSQP as the inner solver for a MIQP, and implement the integer portion as an outer wrapper (so you would call OSQP with your sub-problems at each node).

One such method can be found in some earlier work by @bstellato and @goulart-paul (https://stellato.io/assets/downloads/publications/2018/miosqp_ecc.pdf), and there is some code available for it using Python here: https://github.com/osqp/miosqp.
